reactivesocket-tck-drivers/src/main/java/io/reactivesocket/tckdrivers/common/AddThread.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2016 Facebook, Inc.
+ * <p>
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ *  the License. You may obtain a copy of the License at
+ *  <p>
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  <p>
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivesocket.tckdrivers.common;
+
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * A thread that is created to wait to be able to add a marble string. We wait for the previous thread to have finished
```

Comment:
I don't understand this ... can you elaborate on what a "marble string" is, and why we need a new thread for this?

How many of these threads are running and blocking at any point in time? Do they block the underlying IO?

Extra threads probably doesn't matter here since it's just testing and not prod, but I'd like to confirm.